units/stats_notifications.go

```diff
@@ -0,0 +1,108 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/notification"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/logging"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const (
+	notificationsStatsCollectorJobName = "notifications-stats-collector"
+)
+
+func init() {
+	registry.AddJobType(notificationsStatsCollectorJobName,
+		func() amboy.Job { return makeNotificationsStatsCollector() })
+}
+
+type notificationsStatsCollector struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	logger   grip.Journaler
+}
+
+func makeNotificationsStatsCollector() *notificationsStatsCollector {
+	j := &notificationsStatsCollector{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    notificationsStatsCollectorJobName,
+				Version: 0,
+			},
+		},
+		logger: logging.MakeGrip(grip.GetSender()),
+	}
+	j.SetDependency(dependency.NewAlways())
+	j.SetID(fmt.Sprintf("%s:%s:%d", notificationsStatsCollectorJobName, time.Now().Truncate(0).String(), job.GetNumber()))
+	j.SetPriority(-1)
+	j.UpdateTimeInfo(amboy.JobTimeInfo{
+		MaxTime: time.Minute,
+	})
+
+	return j
+}
+
+func NewNotificationStatsCollector(id string) amboy.Job {
+	job := makeNotificationsStatsCollector()
+	job.SetID(id)
+
+	return job
+}
+
+func (j *notificationsStatsCollector) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	msg := message.Fields{
+		"start_time": j.TimeInfo().Start.String(),
+	}
+
+	e, err := event.FindLastProcessedEvent()
+	j.AddError(errors.Wrap(err, "failed to fetch most recently processed event"))
+	if j.HasErrors() {
+		return
+	}
+	if e != nil {
+		msg["last_processed_at"] = e.ProcessedAt
+	}
+
+	if ctx.Err() == nil {
```

Comment:
I would just check ctx.Err() once in the fuction rather than doing in a bunch of times.

Reply:
I think partially collected data isn't particularly useful (and could happen often enough, as the job's contexts are canceled when the app servers restart.)

it's fine to miss data sometimes.